File: JMP.Enterprises/frontend/src/types/reservation.ts
export interface Reservation {
  reservationId: number;
  propertyId: number;
  propertyName: string;
  propertyCode: string;
  guestId: number;
  guestName: string;
  guestCompanyName?: string | null;
  guestMobileNumber?: string | null;
  rentalType: string; // 'ShortStay' | 'LongStay'
  bookingSource: string; // 'Direct' | 'Airbnb' | 'Facebook' | 'Referral' | 'Other'
  checkInDate: string;
  checkOutDate: string;
  numberOfNights: number;
  dailyRate: number;
  monthlyRate: number;
  agreedRentalAmount: number;
  securityDeposit: number;
  reservationFee: number;
  reservationStatus: string; // 'Inquiry' | 'Reserved' | 'Confirmed' | 'CheckedIn' | 'CheckedOut' | 'Cancelled'
  notes?: string | null;
  createdDate: string;
  updatedDate?: string | null;
}

export interface CreateReservationDto {
  propertyId: number;
  guestId: number;
  rentalType: string;
  bookingSource: string;
  checkInDate: string;
  checkOutDate: string;
  dailyRate: number;
  monthlyRate: number;
  agreedRentalAmount: number;
  securityDeposit: number;
  reservationFee: number;
  reservationStatus: string;
  notes?: string;
}

export interface UpdateReservationDto {
  propertyId: number;
  guestId: number;
  rentalType: string;
  bookingSource: string;
  checkInDate: string;
  checkOutDate: string;
  dailyRate: number;
  monthlyRate: number;
  agreedRentalAmount: number;
  securityDeposit: number;
  reservationFee: number;
  reservationStatus: string;
  notes?: string;
}

export interface PropertyAvailability {
  propertyId: number;
  propertyName: string;
  propertyCode: string;
  isAvailable: boolean;
  currentReservationGuest?: string | null;
  conflictCheckIn?: string | null;
  conflictCheckOut?: string | null;
}

File: JMP.Enterprises/backend/JMP.Enterprises.Api/Models/Reservation.cs
namespace JMP.Enterprises.Api.Models;

/// <summary>
/// Represents a property reservation / booking (Short Stay or Long Stay).
/// Maps to "Reservations" table in SQL Server.
/// </summary>
public class Reservation
{
    public int ReservationId { get; set; }

    public int PropertyId { get; set; }
    public Property? Property { get; set; }

    public int GuestId { get; set; }
    public Guest? Guest { get; set; }

    /// <summary>
    /// ShortStay | LongStay
    /// </summary>
    public string RentalType { get; set; } = "ShortStay";

    /// <summary>
    /// Direct | Airbnb | Facebook | Referral | Other
    /// </summary>
    public string BookingSource { get; set; } = "Direct";

    public DateTime CheckInDate { get; set; }

    public DateTime CheckOutDate { get; set; }

    public decimal DailyRate { get; set; }

    public decimal MonthlyRate { get; set; }

    public decimal AgreedRentalAmount { get; set; }

    public decimal SecurityDeposit { get; set; } = 0;

    public decimal ReservationFee { get; set; } = 0;

    public string? ReservationFeePaymentMethod { get; set; } = "Cash";

    public string? ReservationFeeReferenceNumber { get; set; }

    /// <summary>
    /// Inquiry | Reserved | Confirmed | CheckedIn | CheckedOut | Cancelled
    /// </summary>
    public string ReservationStatus { get; set; } = "Confirmed";

    public string? Notes { get; set; }

    public DateTime CreatedDate { get; set; } = DateTime.Now;

    public DateTime? UpdatedDate { get; set; }
}
